### DISCUSSION TITLE
Having Trouble Understanding Inputs and Outputs with Plugin (Java)

### DISCUSSION THREAD

**Discussion by Maxed0utt** [Jatana/FastOlympicCoding#70] [Q&A]
Hi Friends!

I really can't seem to figure out how to properly use this amazing plugin with Java code!
I don't understand how the inputs and outputs work after I have clicked run on my tests, If anyone can help me understand how to use the plugin properly and what I am doing wrong I would be greatly appreciative! I have attached a screenshot of what I am currently trying when using this plugin.

Thanks!!

![fastolympiccoding_screenshot](https://user-images.githubusercontent.com/95400998/174617379-d7229814-a41d-4978-b150-cb6ee518752a.png)

**Comment by npanuhin** [comment]
@Maxed0utt
Please send the source code here ([gist.github.com](https://gist.github.com/) or [pastebin.com](https://pastebin.com/)). I'm too lazy to retype from a screenshot 🙃

**Reply by Maxed0utt** [reply]
@npanuhin Absolutely, Thank you for your help!
https://gist.github.com/Maxed0utt/3282b2aee2995d26b1e6aca5fdbeb28f

**Comment by npanuhin** [comment]
It looks like you're using [an infinite loop here, collecting numbers from stdin](https://gist.github.com/Maxed0utt/3282b2aee2995d26b1e6aca5fdbeb28f#file-twosum-java-L40-L41). FastOlympicCoding is designed in such a way that it will read UI stdin until it runs out, send it to program's stdin in parallel, print to UI stdout immediately when the program flushes something to its stdout (also in parallel). So what's happening here is that it's just waiting indefinitely for the next number.

- One possible solution is to process numbers immediately after input and print the result to stdout if the task allows it (but it doesn't seem to be the case here)
- The second possible solution is to somehow mark the end of input (`0` marks the end of numbers; the first number specifies the amount of numbers, etc.)

I have never seen a competitive programming problem with a variable amount of input numbers without an end condition. So either write this condition in your code, or output in parallel (if the task is suitable for this).

Imagine that you are running this code in a regular console - try it yourself (`javac TwoSum.java && java TwoSum`), you will get the same endless input.
